R/obic_score.R

```diff
@@ -59,4 +93,62 @@ obic_score <- function(dt.ind) {
 #' This table defines the weighting factors (ranging between 0 and 1) of indicator values to calculate integrated scores. 
 #' This table is used internally in \code{\link{obic_score}}
 #' 
-"weight.obic"
\ No newline at end of file
+"weight.obic"
+
+#' Score the scoring for the OBI on a relative scale
+#' 
+#' This function calculates the relative scoring based on the score calculated by \code{\link{score_absolute}}
+#' 
+#' @param dt.score.abs (data.table) The table containing the score calculated by \code{\link{score_absolute}}
+#' 
+#' @import data.table
+#' 
+#' @export
+score_relative <- function(dt.score.abs) {
+  
+  # Check inputs
+  checkmate::assert_data_table(dt.score.abs)
+  
+  # make local copy
+  dt.score.abs <- copy(dt.score.abs)
+  
+  group_id = V1 = col.sel = waterstress = yield_depression = NULL
+  S_C_R = S_P_R = S_B_R = S_M_R = S_T_R = NULL
+  S_C_A = S_P_A = S_B_A = S_M_A = S_T_A = NULL
+  
+  # Join crop categories
+  crops.obic <- as.data.table(OBIC::crops.obic)
+  col.sel <- c("crop_code", "crop_waterstress")
+  dt.score <- merge(dt.score.abs, crops.obic[, mget(col.sel)], by.x = "B_LU_BRP", by.y = "crop_code", all.x = TRUE)
+
+  # Join the waterstress levels
+  waterstress.obic <- as.data.table(OBIC::waterstress.obic)
+  waterstress.obic[waterstress < 11, yield_depression := 1]
+  waterstress.obic[waterstress >= 11 & waterstress < 30, yield_depression := 2]
+  waterstress.obic[waterstress >= 30, yield_depression := 3]
+  dt.score <- merge(dt.score, waterstress.obic, by.x = c("crop_waterstress", "B_HELP_WENR", "B_GT"), by.y = c("cropname", "soilunit", "gt"))
+
+  # Select columns to base the ranking on
+  grouping <- c("YEAR", "yield_depression", "crop_waterstress", "B_BT_AK")
```

Comment:
is YEAR required here as grouping factor?
probably because the evaluation needs to be done for each year separately?
does that make sense since all these factors (GT, B_HELP_WENR and also waterstress) are quite static variables.
the only thing that varies then is the area of a certain crop in a specific year....

Reply:
Yes, otherwise inflation occurs and comparing with your relative grade for this year with several years age becomes much more difficult